pkg/catalogv2/oci/client.go

```diff
@@ -0,0 +1,307 @@
+			}
+			ociClient.username = username
+			ociClient.password = password
+		} else {
+			return nil, fmt.Errorf("only basicauth credential is supported")
+		}
+	}
+
+	return ociClient, nil
+}
+
+func (o *Client) parseURL() error {
+	// Remove any slash at the end of the URL
+	url := strings.TrimSuffix(o.URL, "/")
+
+	// Remove the oci scheme from the start of the URL
+	url = strings.TrimPrefix(url, "oci://")
+
+	// If the URL contains a slash, then it must have a repository and/or tag
+	if strings.Contains(url, "/") {
+
+		var ref registry.Reference
+		ref, err := registry.ParseReference(url)
+		if err != nil {
+			return fmt.Errorf("failed to parse OCI URL '%s' value: %w", o.URL, err)
+		}
+
+		o.registry = ref.Registry
+		o.repository = ref.Repository
+		o.tag = ref.Reference
+
+		// If the URL doesn't contain any slash, then it must have only the registry part of it.
+	} else {
+		o.registry = url
+	}
+
+	return nil
+}
+
+// fetchChart fetchs the chart specified by the oras repository. It first downloads it into the
+// oras memory and then saves it into a file and returns the file path.
+func (o *Client) fetchChart(orasRepository *orasRegistryRemote.Repository) (string, bool, error) {
+
+	memoryStore := memory.New()
+	manifest, err := oras.Copy(context.Background(), orasRepository, o.tag, memoryStore, "", oras.DefaultCopyOptions)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to oras copy the remote oci artifact %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+	}
+
+	// Fetch the manifest content of the oci artifact
+	manifestblob, err := content.FetchAll(context.Background(), memoryStore, manifest)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to fetch the manifest blob of %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+	}
+
+	var manifestJSON ocispec.Manifest
+	err = json.Unmarshal(manifestblob, &manifestJSON)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to unmarshal manifest blob of %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+	}
+
+	// Create a temp file to store the helm chart tar
+	tempFile, err := os.CreateTemp("", "helm-")
+	if err != nil {
+		return "", false, fmt.Errorf("unable to create temp file for storing the oci artifact")
+	}
+
+	// Check if the oci artifact is of type helm config ?
+	if manifestJSON.ArtifactType == helmregistry.ConfigMediaType || manifestJSON.Config.MediaType == helmregistry.ConfigMediaType {
+		// find the layer of chart type and fetch it
+		for _, layer := range manifestJSON.Layers {
+			if layer.MediaType == helmregistry.ChartLayerMediaType {
+
+				chartTar, err := content.FetchAll(context.Background(), memoryStore, layer)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to fetch chart blob of %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+				}
+
+				err = os.WriteFile(tempFile.Name(), chartTar, 0644)
```

Comment:
Let's use the octal syntax, which highlights the left-most zero is meaningful, and trim the permissions.

```suggestion
				err = os.WriteFile(tempFile.Name(), chartTar, 0o600)
```

Reply:
Done.